apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/SurveyAnalysisCTA.tsx

```diff
@@ -3,99 +3,220 @@
-            }}
-            EndIcon={ArrowUpRightFromSquareIcon}>
-            Preview
-          </Button>
-        </>
+        <Button
+          variant="secondary"
+          size="sm"
+          onClick={() => window.open(getPreviewUrl(), "_blank")}
+          EndIcon={Eye}>
+          Preview
+        </Button>
       )}
+
       {!isViewer && (
         <Button
           size="sm"
-          className="h-full"
           href={`/environments/${environment.id}/surveys/${survey.id}/edit`}
           EndIcon={SquarePenIcon}>
           Edit
         </Button>
       )}
-      {showShareSurveyModal && user && (
-        <ShareEmbedSurvey
-          survey={survey}
-          open={showShareSurveyModal}
-          setOpen={setOpenShareSurveyModal}
-          webAppUrl={webAppUrl}
-          user={user}
-        />
+
+      {!isViewer && (
+        <div id={`${survey.name.toLowerCase().replace(/\s+/g, "-")}-survey-actions`}>
+          <DropdownMenu
+            open={modalState.dropdown}
+            onOpenChange={(open) => setModalState((prev) => ({ ...prev, dropdown: open }))}>
+            <DropdownMenuTrigger className="z-10 cursor-pointer" asChild>
+              <Button variant="secondary" size="sm" className="p-2">
+                <MoreVertical className="h-4 w-4" />
+                <span className="sr-only">Open options</span>
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent className="mr-8 w-40">
+              <DropdownMenuGroup>
+                <DropdownMenuItem asChild>
+                  <button onClick={handleCopyLink} className="flex w-full items-center">
+                    <LinkIcon className="mr-2 h-4 w-4" />
+                    Copy Link
+                  </button>
+                </DropdownMenuItem>
+
+                <DropdownMenuItem asChild>
+                  <button
+                    onClick={() => {
+                      handleModalState("panel")(true);
+                      setModalState((prev) => ({ ...prev, dropdown: false }));
+                    }}
+                    className="flex w-full items-center">
+                    <UsersRound className="mr-2 h-4 w-4" />
+                    Send to panel
+                  </button>
+                </DropdownMenuItem>
+
+                <DropdownMenuItem asChild>
+                  <Link
+                    href={`/environments/${survey.environmentId}/settings/notifications`}
+                    className="flex w-full items-center"
+                    onClick={() => setModalState((prev) => ({ ...prev, dropdown: false }))}>
+                    <BellRing className="mr-2 h-4 w-4" />
+                    Configure alerts
+                  </Link>
+                </DropdownMenuItem>
+              </DropdownMenuGroup>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        </div>
       )}
 
-      {user && <SuccessMessage environment={environment} survey={survey} />}
+      {user && (
```

Comment:
embedding the same modal 3x just because a prop changes is not the right approach. pls refactor :)